functional_tests/steps/methodology_page.py

```diff
@@ -0,0 +1,132 @@
+
+
+@then("the article is displayed correctly under the Related publication section")
+def related_publications_are_displayed_correctly(context: Context):
+    expect(context.page.get_by_role("heading", name="Related publications")).to_be_visible()
+    expect(context.page.locator("li").filter(has_text=f"{context.topic_page.title}")).to_be_visible()
+
+
+@when("the user selects the Contact Details")
+def user_selects_the_contact_details(context: Context):
+    context.page.get_by_role("button", name="Choose contact details").click()
+    context.page.get_by_role("link", name=context.contact_details_snippet.name).click()
+
+
+@then("the Contact Details are visible on the page")
+def contact_details_are_visible_on_the_page(context: Context):
+    # in the header
+    expect(context.page.get_by_text(f"Contact: {context.contact_details_snippet.name}")).to_be_visible()
+    # in the section
+    expect(context.page.get_by_role("heading", name="Contact details")).to_be_visible()
+    expect(context.page.get_by_text(f"Name: {context.contact_details_snippet.name}")).to_be_visible()
+    expect(context.page.get_by_text(f"Email: {context.contact_details_snippet.email}")).to_be_visible()
+
+
+@when("the Last revised date is set to be before the Publication date")
+def set_last_revised_date_before_publication_date(context: Context):
+    context.page.get_by_label("Publication date*").fill("1950-01-02")
+    context.page.get_by_label("Last revised date").fill("1950-01-01")
+
+
+@then("a validation error for the Last revised date is displayed")
+def validation_error_displayed_when_incorrect_date_selected(context: Context):
+    expect(context.page.get_by_text("The last revised date must be after the published date.")).to_be_visible()
+
+
+@then("the mandatory fields raise a validation error")
+def mandatory_fields_raise_validation_error_when_not_set(context: Context):
+    expect(context.page.get_by_text("The page could not be created due to validation errors")).to_be_visible()
+
+    expect(
+        context.page.locator("#panel-child-content-child-title-errors .error-message").get_by_text(
+            "This field is required"
+        )
+    ).to_be_visible()
+
+    expect(
+        context.page.locator("#panel-child-content-child-summary-errors .error-message").get_by_text(
+            "This field is required"
+        )
+    ).to_be_visible()
+
+    expect(
+        context.page.locator(
+            "#panel-child-content-child-metadata-child-panel-child-publication_date-errors .error-message"
+        ).get_by_text("This field is required")
+    ).to_be_visible()
+
+    expect(context.page.locator(".help-block.help-critical").get_by_text("This field is required")).to_be_visible()
+
+
+@then("the preview is visible with the populated data")
+def preview_is_visible(context: Context):
+    context.page.get_by_role("button", name="Toggle preview").click()
+
+    # add a small delay to allow the preview to render
+    context.page.wait_for_timeout(500)
```

Comment:
Can we use an `expect` to wait for an element in the preview, rather than a hardcoded time?

Reply:
Turns out I've prematurely added the delay in this case as the test is passing fine without it - probably the Playwright built-in timeouts are kicking in and waiting for when the preview is visible.